operator/cmd/mesh/manifest-apply.go

```diff
@@ -139,6 +140,21 @@ func InstallManifests(setOverlay []string, inFilenames []string, force bool, dry
 	if err != nil {
 		return err
 	}
+
+	serverVersion, err := clientset.Discovery().ServerVersion()
+	if err != nil {
+		return fmt.Errorf("error getting k8s version: %w", err)
+	}
+	ok, err := k8sversion.CheckKubernetesVersion(serverVersion)
+	if err != nil {
+		return fmt.Errorf("error comparing k8s version: %w", err)
+	}
+	if !ok {
+		return fmt.Errorf(
+			"the kubernetes version is not supported by this version of Istio. The minimum supported kubernetes version is %s",
```

Comment:
Can we add a warning instead? I don't like to have a hard limit without a way to override.

nit: kubernetes/k8s -> Kubernetes

Reply:
+1 on warning instead of error. Regardless this is great, I was so confused the first time it happened to me

Reply:
Does it make sense to have a confirm prompt as well? `Are you sure you want to proceed?(y/N)`?

Reply:
> Does it make sense to have a confirm prompt as well? `Are you sure you want to proceed?(y/N)`?

`istioctl install` already have confirm prompt so I think we don't need another confirm prompt.

https://github.com/istio/istio/blob/0739d2669cd613145c432809f03da5505f94a625/operator/cmd/mesh/manifest-apply.go#L114-L119

Reply:
@shamsher31 yeah, I know that :) I was just wondering, because the user is going to proceed with an unsupported installation he should explicitly opt in for it. Perhaps a simple warning may not caught their attention. Let me change it to a warning and we can easily add a confirmation in a follow-up, if we decide it's needed later.

Reply:
@howardjohn @nschhina @shamsher31 done.